### DISCUSSION TITLE
deglitcher

### DISCUSSION THREAD

**Discussion by mengstr** [hneemann/Digital#1104] [Ideas]
Although the 74193 is a synchronous counter it still have glitches in the simulation (it also may do that on a real device - I haven't tested that yet).  In real life those glitches might be short enough to not trigger following logic, or due to capacitances on the wires never even reach the voltage thresholds.

But when this happens it's not totally uncommon to "fix" this problem with a small lowpass RC-filter on the wire, deleting the rise just a few ns to make the glitch go away.  

In Digital this lowpass filter can easily be emulated with an AND-gate plus a delay element, but it adds clutter to the schematics, so it would be nice to have a really small "deglitcher" component.

A funny thing that happens in the attached design is that the glitch counts (after clicking the button 16 times) are different on each run.
<img width="1579" alt="Screenshot 2023-01-19 at 13 45 09" src="https://user-images.githubusercontent.com/325326/213446543-d39c052a-ab9b-462f-a801-e50d4dda6889.png">

[deglitcher.zip](https://github.com/hneemann/Digital/files/10456767/deglitcher.zip)

**Comment by hneemann** [comment]
This behavior comes from the one gate delay that is missing in the control of the first flip-flop in the 74193. If you add a delay element here, the problem disappears.
Probably this is solved in the original 74193, otherwise one could hardly call the device "synchronous". Therefore I think that one could insert this delay element also in the simulation of the 74193.
